lib/pbench/agent/tool_data_sink.py

```diff
@@ -550,6 +610,117 @@ def terminate(self):
             self.logger.warning("Failed to tar up pmlogger data: %r", args)
 
 
+class BenchmarkRunDir:
+    """BenchmarkRunDir - helper class for handling the benchmark_run_dir
+    directory Redis parameter vs the actual "local" benchmark run directory.
+
+    It is a requirement of the Tool Meister sub-system that the ${pbench_run}
+    directory is always a prefix of the ${benchmark_run_dir}.
+
+    When the pbench CLI starts the Tool Data Sink directly, the local
+    benchmark run directory is the same as the value of the benchmark_run_dir
+    parameter.
+
+    But when the Tool Data Sink runs in a container, the local volume name for
+    the benchmark run directory might be different from the parameter value.
+    Typically, the container is constructed with the default pbench
+    installation, where the ${pbench_run} directory is "/var/lib/pbench-agent".
+
+    The entity responsible for starting the Tool Data Sink container typically
+    mounts a different directory for /var/lib/pbench-agent via 'podman run
+    --volume /srv/data/pbench-run-dir:/var/lib/pbench-agent:Z'.  This leads to
+    a conflict where the external ${pbench_run} path is different from the
+    internal to the container ${pbench_run} path.  To resolve this, the entity
+    which creates the external pbench run directory creates a ".path" file in
+    that directory containing the full "external" path to the pbench run
+    directory. The Tool Data Sink uses that path to validate that external
+    benchmark_run_dir parameter values are valid.
+
+    This class implements the mechanism that allows the Tool Data Sink code to
+    handle that seamlessly.
+    """
+
+    def __init__(self, ext_benchmark_run_dir, int_pbench_run):
+        self._ext_benchmark_run_dir = Path(ext_benchmark_run_dir)
+        self._ext_pbench_run = self._ext_benchmark_run_dir.parent
+        self._int_pbench_run = Path(int_pbench_run)
+
+        # The Tool Data Sink could be running in a container. If so, then
+        # it'll be using the default benchmark run directory.  If the
+        # benchmark_run_dir parameter is valid, there will be a file
+        # called ".path" in the default benchmark run directory which will
+        # match.
+        #
+        # E.g.:
+        #  $ pbench_run="/home/<USER>/run-dir"
+        #  $ benchmark_run_dir="${pbench_run}/script_config_<date>"
+        #  $ cat ${pbench_run}/.path
+        #  /home/<USER>/run-dir
```

Comment:
Was trying to show the output instead.

Reply:
My point was that it appears that we know what the contents of the `.path` file should be -- namely, it should match the contents of the `pbench_run` environment variable; so, rather than showing what the contents of the file should be, we can just put the corrent contents into it...thereby removing a point of failure.

Reply:
As we discussed in realtime, in this example use case description, we have two sources for a common truth -- we have the `pbench_run` variable which is used in the `podman` invocation, and we have the contents of the `.path` file which is used inside the container.  Since they both need to contain the same value, we should suggest that one be derived from the other (either by using the command I suggested above to generate the `.path` file from the variable's value, or by using the contents of the file to define the variable or using it directly on the `podman` command line).